### PR TITLE
Add ability to throw error

### PR DESCRIPTION
**Pull request by joesonw**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,6 +55,9 @@ function ratelimit(opts) {
 
     var delta = (limit.reset * 1000) - Date.now() | 0;
     var after = limit.reset - (Date.now() / 1000) | 0;
+    if (opts.throwable) {
+      throw opts.throwable(after,ms(delta, { long: true }));
```
Comment (dead-horse) [inline index.js]: it is better to throw an error like:

```
this.throw(429, 'Rate limit exceeded, retry in ' + ms(delta, { long: true }), {
  after: after,
  delta: delta,
  code: RATE_LIMIT
})
```

and you can catch this error outside.

Comment (dead-horse) [inline index.js]: also you should add a test case for this

Comment (joesonw) [inline index.js]: this allows for customized errors.in case you do not want to re-throw another error to meet application's standards

Comment (dead-horse) [inline index.js]: or something like this:

```js
function ratelimit(opts) {
  var handler = opts.limitHandler || defaultLimitHandler;
//...
  return function* (next) {
    // ...
    handler(ctx, after, delta)
  }


function defaultLimitHandler(ctx, after, delta) {
  ctx.set('Retry-After', after);
  ctx.status = 429;
  ctx.body = 'Rate limit exceeded, retry in ' + ms(delta, { long: true });
}
```

**Comment by ruimarinho** [comment]
Isn't this the same as #14?

**Pull request closed without merging**